File: MatchAnalyzer.Api/Models/Match.cs
namespace MatchAnalyzer.Api.Models;

public class Match
{
    public int Id { get; set; }
    public string HomeTeam { get; set; } = string.Empty;
    public string AwayTeam { get; set; } = string.Empty;
    public DateTime? Date { get; set; }
    public string? Score { get; set; }
    public int? GoalsCount { get; set; }
    public int? FirstHalfGoals { get; set; }
    public bool IsParsed { get; set; }
    
    public string MatchId { get; set; } = string.Empty;
    
    public string HomeTeamId { get; set; } = string.Empty;
    public string AwayTeamId { get; set; } = string.Empty;
}

File: MatchAnalyzer.Web/src/app/models/match.ts
export interface Match {
  id: number;
  matchId: string;
  homeTeam: string;
  awayTeam: string;
  date: string; // ISO string
  score?: string;
  goalsCount?: number;
  firstHalfGoals?: number;
  isParsed: boolean;
  isFavorite05: boolean;
  isFavorite15: boolean;
  isFavoriteFH05: boolean;
  isFavoriteFH15: boolean;
}
